File: sdk/python/tests/test_arv_ws.py
# Copyright (C) The Arvados Authors. All rights reserved.
#
# SPDX-License-Identifier: Apache-2.0

import os
import sys
import tempfile
import unittest

import arvados.errors as arv_error
import arvados.commands.ws as arv_ws
from . import arvados_testutil as tutil

class ArvWsTestCase(unittest.TestCase):
    def run_ws(self, args):
        return arv_ws.main(args)

    def test_ctrl_c(self):
        with (
            self.assertRaises(SystemExit) as cm,
            unittest.mock.patch(
                "arvados.events.EventClient.run_forever",
                unittest.mock.Mock(side_effect=KeyboardInterrupt)
            )
        ):
            self.run_ws([])
        self.assertEqual(cm.exception.code, 0)

File: sdk/cli/test/test_arv-ws.rb
# Copyright (C) The Arvados Authors. All rights reserved.
#
# SPDX-License-Identifier: Apache-2.0

require 'minitest/autorun'

class TestArvWs < Minitest::Test
  def setup
  end

  def test_arv_ws_get_help
    out, err = capture_subprocess_io do
      system ('arv-ws -h')
    end
    assert_equal '', err
  end

  def test_arv_ws_such_option
    out, err = capture_subprocess_io do
      system ('arv-ws --junk')
    end
    refute_equal '', err
  end

end
